src/hooks/useContentProtection.ts

```diff
@@ -31,13 +33,15 @@ const useContentProtection = <T>(
     async () => {
       // if provider is not JWP
       if (!!id && !!host) {
-        const authData = await accountService.getAuthData();
+        const accountController = getModule(AccountController);
+        const authData = await accountController.getAuthData();
         const { host, drmPolicyId } = signingConfig;
-        return getMediaToken(host, id, authData?.jwt, params, drmPolicyId);
+
+        return entitlementController.getMediaToken(host, id, authData?.jwt, params, drmPolicyId);
       }
       // if provider is JWP
       if (jwp && configId && !!id && signingEnabled) {
-        return getJWPMediaToken(configId, id);
+        return entitlementController.getJWPMediaToken(configId, id);
```

Comment:
Kind of weird that entitlementController has 2 separate methods that are integration aware. Is there a way to abstract this so there's just a getMediaToken method that does the right logic with the right dependencies?

Reply:
I think that was my mistake to use `getMediaToken` for both controller methods.
Looks like `getMediaToken` is a legacy method (`getJWPMediaToken` is not), so maybe it is also better to keep them separate.

Reply:
Fixed